### ISSUE TITLE
Autogen struct raises Dry::Struct::Error on Timestamp attribute

### ISSUE THREAD

**Issue by abrthel** [rom-rb/rom-repository#57]
Here's a weird issue, I have a relation and repository that looks similar to the ones defined below. If I try and update a person from a repository, I get a `Dry::Struct::Error`. 

I will provide this for now and hopefully later today I can get time to create a script that fully reproduces the error.

```ruby
class People < ROM::Relation[:sql]
  schema(:people, infer: true) do
    attribute :id, Types::Strict::Int.meta(primary_key: true)
    attribute :fullname, Types::Strict::String.optional
    attribute :birth_date, Types::Strict::Date.optional
    attribute :created_at, Types::Time
    attribute :updated_at, Types::Time
    attribute :ended_at,  Types::Time.optional
  end
end

class PeopleRepository < ROM::Repository[:people]
  commands :create, update: :by_pk, delete: :by_pk
  def by_id(id)
    people.fetch(id)
  end
end

repo = PersonRepository.new(romcontainer)
repo.update(1, fullname: 'Blah Blah')
```
```
Dry::Struct::Error: [ROM::Struct[Person].new] 2017-02-01 22:22:04 +0000 (Time) has invalid type for :created_at
        from /usr/local/rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/dry-struct-0.1.1/lib/dry/struct/class_interface.rb:80:in `rescue in new'
        from /usr/local/rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/dry-struct-0.1.1/lib/dry/struct/class_interface.rb:74:in `new'
        from /usr/local/rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/transproc-1.0.0/lib/transproc/class.rb:30:in `constructor_inject'
        (...)
```
The value of `:created_at` for the row in question is `2017-02-01 22:22:04.877173` and the db is postgres

The relation links to an updateable view composed of two separate tables where triggers are defined to do the actual updating (they've been omitted since I know they work)
```sql
CREATE TABLE parties
(
  id integer NOT NULL DEFAULT,
  fullname text
  created_at timestamp without time zone NOT NULL DEFAULT now(),
  updated_at timestamp without time zone NOT NULL DEFAULT now(),
  ended_at timestamp without time zone
)

CREATE TABLE party_people
(
  id integer NOT NULL,
  birth_date date
)

CREATE OR REPLACE VIEW people AS 
 SELECT t1.id,
    t1.fullname,
    t2.birth_date,
    t1.created_at,
    t1.updated_at,
    t1.ended_at
   FROM parties t1
     JOIN party_people t2 ON t1.id = t2.id;
```

Environment:
```
database: PostgreSQL 9.6rc1

gems:
pg (0.19.0)
rom (3.0.1)
rom-mapper (0.5.0)
rom-model (0.5.0)
rom-rails (0.9.0)
rom-repository (1.0.1)
rom-sql (1.0.0)
rom-support (2.0.0)
```

**Comment by solnic** [comment]
does it work with inferred schema? just do `schema(infer: true)`?

**Comment by abrthel** [comment]
Nope, same issue.

**Comment by solnic** [comment]
OK I'll look into that. Thanks for reporting it!

**Comment by abrthel** [comment]
Ok, so I spent the better part of 2hrs trying to write a script which reproduces the error and nothing... it worked every time. 

So I figured it was my rails environment that was misconfigured. I loaded the environment in my script and repository creations and updates on the people view began to work just like in my test script. I then tried the exact same update commands in a rails console and it failed just like in my bug report. So I went from everything working in a test script to failing only in a rails console.

Test Script: [gist](https://gist.github.com/abrthel/c92bc9d219219d53612815679a21e73c)

I think the actual issue was related to spring in some way because after a complete reset of my environment, running the rails console and my script work and I can't reproduce the error anymore. Honestly I have no clue what happened or why it failed in the first place. For now we can close this.

One thing I did learn, on views the schema must have the primary key defined, it can't be inferred. Otherwise `people.update(1, fullname: 'James')` raises

```
/usr/local/rbenv/versions/2.3.3/lib/ruby/gems/2.3.0/gems/rom-3.0.1/lib/rom/schema.rb:164:in `[]': nil attribute doesn't exist in people schema (KeyError)
```

which I suppose makes sense since the view itself doesnt have a primary key. Maybe a new issue for a better error message?
